src/databricks/labs/ucx/aws/access.py

```diff
@@ -296,18 +296,24 @@ def get_instance_profile(self, instance_profile_name: str) -> AWSInstanceProfile
 
         return AWSInstanceProfile(instance_profile_arn)
 
-    def _create_uber_instance_profile(self, iam_role_name):
-        self._aws_resources.create_migration_role(iam_role_name)
-        self._aws_resources.create_instance_profile(iam_role_name)
+    def _create_uber_instance_profile(self, iam_role_name: str, iam_policy_name: str, s3_paths: set[str]):
         # Add role to instance profile - they have the same name
-        self._aws_resources.add_role_to_instance_profile(iam_role_name, iam_role_name)
+        if (
+            not (self._aws_resources.create_migration_role(iam_role_name))
+            or not (self._aws_resources.create_instance_profile(iam_role_name))
+            or not (self._aws_resources.add_role_to_instance_profile(iam_role_name, iam_role_name))
+            or not (
+                self._aws_resources.put_role_policy(
+                    iam_role_name, iam_policy_name, s3_paths, self._aws_account_id, self._kms_key
+                )
+            )
```

Comment:
Remove redundant parenthesis:

```suggestion
            not self._aws_resources.create_migration_role(iam_role_name)
            or not self._aws_resources.create_instance_profile(iam_role_name)
            or not self._aws_resources.add_role_to_instance_profile(iam_role_name, iam_role_name)
            or not self._aws_resources.put_role_policy(
                    iam_role_name, iam_policy_name, s3_paths, self._aws_account_id, self._kms_key
                )
```

Reply:
OK